### DISCUSSION TITLE
Looking For a Challenge? Problem in Part 3

### DISCUSSION THREAD

**Discussion by ZimmZone** [earthlab-education/Earth-Analytics-AY24#355] [Q&A]
Good evening everyone! Is anyone able to help me work through the 'Looking for an Extra Challenge?' task from part 3 on the Climate Coding Challenge. 

Here are my initial questions:

1. Can I just change the name, or do I need to defined it like old_name = new_name?
2. What are we supposed to do with the '''Convert temperature to Celsius'''? Is this where we put the equation and then put it again in line three?

**Comment by eculler** [comment]
@ZimmZone  great questions!

1. You CAN use the same name, and you'll see me do that sometimes. But I don't generally recommend it when you're getting started with coding in interactive notebooks (like our .ipynb files) because it is really easy to end up in a bit of a pickle from accidentally running code out of order. 
2. So, that line is what we call a **docstring** in Python. All functions should have them, and they tell folks reading your code what the function does. Much more on that later when we do functions in detail! So, where should your code go? It should go underneath the docstring, but still indented (reducing the indent tells Python you're done with the function). In this case, you only have one line of code, so it can go right after `return`, but usually you'll have more code and you might want to define some variables inside the function and return one of them.

We'll talk about functions in a lot of detail later, so don't feel like you have to complete this extra challenge before moving on. But if you are looking to spice things up with the extra challenge, you can also search for one of the many examples on the internet, or ask ChatGPT about it, or continue asking questions here!
